File: app/lib/models/commerce/review.dart
class Review {
  final String fromUserId;
  final String toUserId;
  final String relatedService;
  final String relatedId;
  final int rating;
  final String comment;
  final DateTime createdAt;

  Review({
    required this.fromUserId,
    required this.toUserId,
    required this.relatedService,
    required this.relatedId,
    required this.rating,
    required this.comment,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) => Review(
        fromUserId: json['from_user_id'],
        toUserId: json['to_user_id'],
        relatedService: json['related_service'],
        relatedId: json['related_id'],
        rating: json['rating'],
        comment: json['comment'],
        createdAt: DateTime.parse(json['created_at']),
      );

  Map<String, dynamic> toJson() => {
        'from_user_id': fromUserId,
        'to_user_id': toUserId,
        'related_service': relatedService,
        'related_id': relatedId,
        'rating': rating,
        'comment': comment,
        'created_at': createdAt.toIso8601String(),
      };
}

File: functions/entity/review.js
// reviews 컬렉션 (후기 및 평점)
// 문서 ID: reviewId (자동 생성)
// {
//   from_user_id: string,
//   to_user_id: string,
//   related_service: 'guide' | 'purchase',
//   related_id: string,
//   rating: number,
//   comment: string,
//   created_at: timestamp
// }

class Review {
  constructor(data) {
    this.from_user_id = data.from_user_id;
    this.to_user_id = data.to_user_id;
    this.related_service = data.related_service;
    this.related_id = data.related_id;
    this.rating = data.rating;
    this.comment = data.comment;
    this.created_at = data.created_at;
  }
}
module.exports = Review;
